packages/dm-core-plugins/src/explorer/components/dialogs/NewEntityDialog.tsx

```diff
@@ -57,13 +62,24 @@ const NewEntityDialog = (props: TProps) => {
       <Dialog.CustomContent>
         <BlueprintPicker
           label='Blueprint'
-          onChange={setBlueprint}
-          formData={blueprint}
+          onChange={setBlueprintName}
+          formData={blueprintName}
           fieldType={'input-field'}
         />
+        {blueprint?.attributes.filter(
+          (attr: TAttribute) => attr.name === 'name'
+        ) && (
+          <TextField
+            id={'entity-name'}
+            onChange={(e: any) => setNewName(e.target.value)}
+            label={'New name'}
+            helperText={'Optional'}
```

Comment:
How do you know it's optional? Should check the blueprint attribute

Reply:
Or just remove it since you set a default name anyway

Reply:
Added a check to see if it is required 👍